metrics/ci_integrations/lib/cli/parties/configured_parties/configured_party.dart

```diff
@@ -0,0 +1,27 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:ci_integration/integration/interface/base/config/model/config.dart';
+import 'package:ci_integration/integration/interface/base/party/integration_party.dart';
+
+/// An abstract class that represents a [Config] and an [IntegrationParty]
+/// that accepts this [Config].
+abstract class ConfiguredParty<T extends Config, P extends IntegrationParty> {
```

Comment:
Would it be great to annotate this abstract class as `immutable`? 
```suggestion
/// that accepts this [Config].
@immutable
abstract class ConfiguredParty<T extends Config, P extends IntegrationParty> {
```